### PR TITLE
Disable Story posts when Jetpack features are removed

### PR DESCRIPTION
**Pull request by fluiddot**
Disables the option to create a Story post when the Jetpack-powered features are removed.

Ref: pe7hp4-4k-p2

**To test:**
### 1 - Story post **ENABLED** in **WordPress app**

1. Navigate to App Settings in the WordPress app and open Debug settings.
2. Turn `Jetpack Features Removal Phase Four` and `Jetpack Features Removal Phase For New Users` flags off.
3. Navigate to My Site screen.
4. Tap on the floating button located in the low-right corner.
5. Observe that the Story post option is displayed.
6. Navigate to the Posts list screen
7. Tap on the floating button located in the low-right corner.
8. Observe that the Story post option is displayed.

### 2 - Story post **DISABLED** in **WordPress app**

1. Navigate to App Settings in the WordPress app and open Debug settings.
2. Turn `Jetpack Features Removal Phase Four` flag on.

**NOTE:** It's recommended that the following steps are also checked by turning the `Jetpack Features Removal Phase For New Users` flag on and  `Jetpack Features Removal Phase Four` flag off in Debug settings.

3. Navigate to My Site screen.
**NOTE:** If you were on the My Site screen when changing the flags, you might need to navigate to a different tab and back again to My Site to see the changes reflected.
4. Tap on the floating button located in the low-right corner.
5. Observe that the Story post option is NOT displayed.
6. Navigate to the Posts list screen
7. Tap on the floating button located in the low-right corner.
8. Observe that no option is displayed and that the editor is opened.

### 3- Story post is **ENABLED** in **Jetpack app**

1. Navigate to App Settings in the Jetpack app and open Debug settings.
2. Turn `Jetpack Features Removal Phase Four` flag on. _This flag should be omitted by the Jetpack app._
3. Navigate to My Site screen.
4. Tap on the floating button located in the low-right corner.
5. Observe that the Story post option is displayed.
6. Navigate to the Posts list screen
7. Tap on the floating button located in the low-right corner.
8. Observe that the Story post option is displayed.

## Regression Notes
1. Potential unintended areas of impact
The changes should only impact the floating Create button located on My Site and Posts list screens.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/System/Floating Create Button/SheetActions.swift [hunk 1]
```diff
@@ -75,4 +75,8 @@ struct StoryAction: ActionSheetItem {
         badge.backgroundColor = Constants.Badge.backgroundColor
         return badge
     }
+
+    static func shouldBeShown(blog: Blog?) -> Bool {
+        return (blog?.supports(.stories) ?? false) && !UIDevice.isPad() && !JetpackFeaturesRemovalCoordinator.shouldRemoveJetpackFeatures()
+    }
```
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/System/Floating Create Button/SheetActions.swift]: I noticed that we weren't using the same conditions for determining if the Story Post action should be shown in different places. For this reason, I decided to unify the logic into a single function. Specifically, checking if the device is an iPad. Based on [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15996) and [this Kanvas issue](https://github.com/tumblr/kanvas-ios/issues/104), I understand that the Story post shouldn't be enabled on iPad.

File: WordPress/Classes/Models/Blog.m [hunk 2]
```diff
@@ -691,7 +691,8 @@ - (BOOL)supportsPluginManagement
 - (BOOL)supportsStories
 {
     BOOL hasRequiredJetpack = [self hasRequiredJetpackVersion:@"9.1"];
-    return hasRequiredJetpack || self.isHostedAtWPcom;
+    // Stories are disabled in iPad until this Kanvas issue is solved: https://github.com/tumblr/kanvas-ios/issues/104
+    return (hasRequiredJetpack || self.isHostedAtWPcom) && ![UIDevice isPad] && ![JetpackFeaturesRemovalCoordinator shouldRemoveJetpackFeatures];
```
Comment (fluiddot) [inline WordPress/Classes/Models/Blog.m]: I noticed that we weren't using the same conditions for determining if the Story Post action should be shown in different places. For this reason, I decided to unify the logic into a single function. Specifically, checking if the device is an iPad. Based on [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15996) and [this Kanvas issue](https://github.com/tumblr/kanvas-ios/issues/104), I understand that the Story post shouldn't be enabled on iPad.

File: WordPress/Classes/Models/Blog.m [hunk 3]
```diff
@@ -691,7 +691,8 @@ - (BOOL)supportsPluginManagement
 - (BOOL)supportsStories
```
Comment (fluiddot) [inline WordPress/Classes/Models/Blog.m:691]: I originally considered checking the `isPad` and feature removal conditions in a specific helper within the [`StoryAction` class](https://github.com/wordpress-mobile/WordPress-iOS/blob/3eb3666f8cf358292aac1ace26b58dd694294d04/WordPress/Classes/ViewRelated/System/Floating%20Create%20Button/SheetActions.swift#L41). But I finally ended up using the `supports` functionality of the Blog model as the logic seemed closer to me. In any case, please let me know if there's a better approach for this, thanks!

File: WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift [hunk 4]
```diff
@@ -108,6 +108,7 @@ class JetpackFeaturesRemovalCoordinator {
     }
 
     /// Used to determine if the Jetpack features should be removed based on the removal phase.
+    @objc
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift:111]: Regarding `JetpackFeaturesRemovalCoordinator.shouldRemoveJetpackFeatures()`
I've removed this function in #19817 based on this discussion https://github.com/wordpress-mobile/WordPress-iOS/pull/19816#discussion_r1058906631 and replaced it with `JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()`

Let me know if you need help addressing this conflict or if you think we should re-add `shouldRemoveJetpackFeatures` 👍

**Review by hassaanelgarem — APPROVED** [review]
Works as described! 🚀

File: WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift (hunk 4 above)
Comment (fluiddot) [reply WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift:111]: Ok no problem, I'll update this PR with that change to solve the conflicts.

> Let me know if you need help addressing this conflict or if you think we should re-add shouldRemoveJetpackFeatures 👍

Thanks! Let's stick with the new function `JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()`, it makes more sense as for most of the conditions I had to negate the value when using `shouldRemoveJetpackFeatures()` 😅 .

Comment (fluiddot) [reply WordPress/Classes/ViewRelated/Jetpack/Branding/Coordinator/JetpackFeaturesRemovalCoordinator.swift:111]: Conflicts solved in https://github.com/wordpress-mobile/WordPress-iOS/pull/19823/commits/6336f6260ae944790b94a5be202bb71563ffcd9c.

**Pull request merged**